azext_iot/operations/hub.py

```diff
@@ -2206,14 +2206,13 @@ def http_wrap(target, device_id, generator):
 
     try:
         if protocol_type == ProtocolType.mqtt.name:
-            wrap = mqtt_client_wrap(
-                target=target,
-                device_id=device_id,
-                properties=properties_to_send,
-                sas_duration=(msg_count * msg_interval)
-                + 60,  # int type is enforced for msg_count and msg_interval
-            )
-            wrap.execute(generator(), publish_delay=msg_interval, msg_count=msg_count)
+            device_connection = iot_get_device_connection_string(cmd=cmd, device_id=device_id, hub_name=hub_name, login=login)
+            client_mqtt = mqtt_client(
+                            target=target,
+                            device_connection_string=device_connection["connectionString"],
+                            device_id=device_id
+                        )
+            client_mqtt.execute(data=generator(), properties=properties_to_send, publish_delay=msg_interval, msg_count=msg_count)
         else:
             six.print_("Sending and receiving events via https")
```

Comment:
Does the python device client also support http? If so we need to align the http behavior as well.

Reply:
HTTP is not supported by the Python device client